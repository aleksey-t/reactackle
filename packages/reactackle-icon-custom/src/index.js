import { iconCustomSizeMixin } from './iconCustomSizeMixin';
import { IconCustom } from './IconCustom';

export {
  IconCustom,
  iconCustomSizeMixin,
};
