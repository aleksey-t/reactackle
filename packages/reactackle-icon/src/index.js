import Icon from './Icon';

export { Icon };
export * from './iconPropType';
export * from './iconStyleMixin';
