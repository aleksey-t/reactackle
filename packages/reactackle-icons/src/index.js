import { registerDefaultComponentTheme } from 'reactackle-core';
import componentTheme from './styles/theme';

registerDefaultComponentTheme('icon', componentTheme);

export * from './IconCustom';
export * from './IconSvg';
export * from './iconStyleMixin';
export * from './iconsPropType';
