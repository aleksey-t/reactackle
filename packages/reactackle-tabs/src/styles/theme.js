import {
  bmodule,
  colorTransparent,
  colorMain,
  bodyFontColor,
  bodyFontColorAlt,
  fontSizeBody,
  fontWeightNormal,
  fontWeightSemibold,
} from 'reactackle-core';

export default {
  tabs: {
    style: {
      light: {
        backgroundColor: colorTransparent,
      },

      dark: {
        backgroundColor: colorTransparent,
      },
    },
  },

  tab: {
    fontSize: fontSizeBody,
    lineHeight: 1.25,
    textTransform: 'uppercase',
    paddingX: bmodule(1.5),
    paddingY: bmodule(1),
    minHeight: bmodule(6),
    minWidth: 120,
    maxWidth: 320,

    style: {
      light: {
        fontColor: bodyFontColorAlt,
        backgroundColor: colorTransparent,
        fontWeight: fontWeightNormal,
        opacity: 0.8,

        hover: {
          fontColor: bodyFontColorAlt,
          backgroundColor: 'rgba(255, 255, 255, 0.1)',
          fontWeight: fontWeightNormal,
          opacity: 0.9,
        },

        focus: {
          fontColor: bodyFontColorAlt,
          backgroundColor: 'rgba(255, 255, 255, 0.1)',
          fontWeight: fontWeightNormal,
          opacity: 1,
        },

        selected: {
          fontColor: bodyFontColorAlt,
          backgroundColor: colorTransparent,
          fontWeight: fontWeightSemibold,
          opacity: 1,
        },
      },

      dark: {
        fontColor: bodyFontColor,
        backgroundColor: colorTransparent,
        fontWeight: fontWeightNormal,
        opacity: 0.8,

        hover: {
          fontColor: bodyFontColor,
          backgroundColor: 'rgba(0, 0, 0, 0.1)',
          fontWeight: fontWeightNormal,
          opacity: 0.9,
        },

        focus: {
          fontColor: bodyFontColor,
          backgroundColor: 'rgba(0, 0, 0, 0.1)',
          fontWeight: fontWeightNormal,
          opacity: 1,
        },

        selected: {
          fontColor: bodyFontColor,
          backgroundColor: colorTransparent,
          fontWeight: fontWeightSemibold,
          opacity: 1,
        },
      },
    },
  },

  icon: {
    height: 40,
    width: null,
    imgSize: null,
    iconTextSpacing: bmodule(1),

    style: {
      light: {
        color: 'inherit',

        hover: {
          color: 'inherit',
        },

        focus: {
          color: 'inherit',
        },

        selected: {
          color: 'inherit',
        },
      },

      dark: {
        color: 'inherit',

        hover: {
          color: 'inherit',
        },

        focus: {
          color: 'inherit',
        },

        selected: {
          color: 'inherit',
        },
      },
    },
  },

  bar: {
    thickness: 4,

    style: {
      light: {
        backgroundColor: colorMain,
      },

      dark: {
        backgroundColor: colorMain,
      },
    },
  },
};
