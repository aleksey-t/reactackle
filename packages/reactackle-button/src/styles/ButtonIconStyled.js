import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { extractThemeOrDefault, getValueString } from 'reactackle-core';
import { iconSvgSizeMixin } from 'reactackle-icon-svg';
import { iconCustomSizeMixin } from 'reactackle-icon-custom';

const propTypes = {
  /** Define button's size */
  size: PropTypes.oneOf(['inline', 'small', 'normal', 'large']),
  /** Swap icon and text position */
  iconPositionRight: PropTypes.bool,
};

const defaultProps = {
  size: 'normal',
  iconPositionRight: false,
};

// PROP RECEIVERS
const iconStyleProps = ({
  theme: themeFromProvider,
}) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const path = theme.reactackle.components.button;
  const opacityAmount = path.iconOpacity;

  return css`
    opacity: ${opacityAmount};   
    
    svg {
      fill: currentColor;
    }
  `;
};

const iconSizeProps = ({ size, type, theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const path = theme.reactackle.components.button.size[size].icon;
  const sizeMixin = type === 'svg' ? iconSvgSizeMixin : iconCustomSizeMixin;

  return css`
    ${sizeMixin(
      getValueString(path.width),
      getValueString(path.imgSize || path.width),
      getValueString(path.height || path.width),
    )}
    
    ${size === 'inline' && `
      display: inline-flex;
      align-self: center;
    `}
  `;
};

export const ButtonIconBoxStyled = styled.div`
  order: ${props => (props.iconPositionRight ? '2' : '1')};
  display: flex;
  ${iconStyleProps};
  ${iconSizeProps};
`;

ButtonIconBoxStyled.propTypes = propTypes;
ButtonIconBoxStyled.defaultProps = defaultProps;
ButtonIconBoxStyled.displayName = 'ButtonIconBoxStyled';
