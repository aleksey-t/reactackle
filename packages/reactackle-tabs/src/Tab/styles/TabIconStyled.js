import styled, { css } from 'styled-components';
import { iconStyleMixin } from 'reactackle-icons';
import {
  extractThemeOrDefault,
  getValueString,
  iconSizeMixin,
} from 'reactackle-core';

const size = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const { width, height, imgSize } = theme.reactackle.components.tabs.icon;

  return css`    
    ${iconSizeMixin(
      getValueString(height || width),
      getValueString(imgSize || height),
      getValueString(width || height),
    )}
  `;
};

const style = ({ theme: themeFromProvider, selected, colorScheme }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const source = theme.reactackle.components.tabs.icon.style[colorScheme];

  return !selected
    ? css`
      ${iconStyleMixin(source.color)};
      
      &:hover {
        color: ${source.hover.color};
      }
      
      &:focus {
        color: ${source.focus.color};
      }
    `
    : css`      
      &,
      &:hover,
      &:focus {
        ${iconStyleMixin(source.selected.color)};        
      }
    `;
};

export const TabIconStyled = styled.div`
  margin-left: auto;
  margin-right: auto;
  display: flex;
  ${size}
  ${style};
`;

TabIconStyled.displayName = 'TabIconStyled';
