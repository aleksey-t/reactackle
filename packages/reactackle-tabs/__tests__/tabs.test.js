/* eslint-disable react/prop-types */
import React from 'react';
import renderer from 'react-test-renderer';
import { findAllByType, setProps } from 'reactackle-test-utils';
import { Tabs } from '../src';
import { TabStyled } from '../src/Tab/styles/TabStyled';

jest.mock('react-dom');

describe('<Tabs/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(<Tabs />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop selected set 1', () => {
    const tree = renderer.create(
      <Tabs
        selected={1}
        tabs={[
        { text: 'Test' },
        { text: 'Test' },
        { text: 'Test' },
        ]}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop linkComponent passed', () => {
    const LinkComponent = props => {
      const content = props.children || props.title;
    
      return (
        <a href={props.href} onClick={props.onClick}>
          {content}
        </a>
      );
    };
    const tree = renderer.create(
      <Tabs
        linkComponent={LinkComponent}
        tabs={[
        { text: 'Test', linkHref: '/' },
        ]}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop icon', () => {
    const tree = renderer.create(
      <Tabs
        tabs={[
          { text: 'Test', icon: { type: 'svg' } },
        ]}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop text is empty', () => {
    const tree = renderer.create(
      <Tabs
        tabs={[
          { text: '' },
        ]}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop ', () => {
    const tree = renderer.create(
      <Tabs
        tabs={[
          { text: 'Test1', linkHref: 'test1' },
          { text: 'Test2', linkHref: 'test2' },
        ]}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if change tab', () => {
    const mockFn = jest.fn(),
      component = renderer.create(
        <Tabs
          onChange={mockFn}
          tabs={[
            { text: 'Test 1', icon: { type: 'svg' } },
            { text: 'Test 2', icon: { type: 'svg' } },
          ]}
        />,
      );

    expect(component.toJSON()).toMatchSnapshot();

    const tabs = findAllByType(component, TabStyled);
    tabs[1].props.onClick();

    expect(mockFn).toHaveBeenCalledWith({ value: 1 });
    expect(component.toJSON()).toMatchSnapshot();
  });


  it('renders correctly if prop selected set after render', () => {
    const tree = renderer.create(
      <Tabs
        selected={2}
        tabs={[
        { text: 'Test' },
        { text: 'Test' },
        { text: 'Test' },
        ]}
      />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
    setProps(tree, { selected: 1 });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
