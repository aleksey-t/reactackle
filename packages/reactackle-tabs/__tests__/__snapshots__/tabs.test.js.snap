// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs/> renders correctly if change tab 1`] = `
<div
  className="TabsWrapperStyled__TabsWrapperStyled-kxWKwf kQdeyK"
>
  <div
    className="TabsContentWrapperStyled__TabsContentWrapperStyled-fmMkZg HDXly"
  >
    <div
      className="TabsListStyled__TabsListStyled-jltvsP iRzhNT"
    >
      <div
        className="TabStyled__TabStyled-kKgjJZ jfjewJ"
        onClick={[Function]}
        selected={true}
      >
        <div
          className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
        >
          <div
            className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
          >
            <div
              className="TabIconStyled__TabIconStyled-hvQHiJ fUMIYU"
              selected={true}
              type="svg"
            >
              <div
                className="IconSvgStyled__IconSvgStyled-iCtyLo hlSHcB"
                color=""
                rotate={0}
                size="normal"
                type="svg"
              >
                <svg
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="TabTextStyled__TabTextStyled-jbYAge hCrmvf"
            >
              Test 1
            </div>
          </div>
        </div>
      </div>
      <div
        className="TabStyled__TabStyled-kKgjJZ jFBgwR"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
        >
          <div
            className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
          >
            <div
              className="TabIconStyled__TabIconStyled-hvQHiJ kYPJWE"
              selected={false}
              type="svg"
            >
              <div
                className="IconSvgStyled__IconSvgStyled-iCtyLo hlSHcB"
                color=""
                rotate={0}
                size="normal"
                type="svg"
              >
                <svg
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="TabTextStyled__TabTextStyled-jbYAge hCrmvf"
            >
              Test 2
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly if change tab 2`] = `
<div
  className="TabsWrapperStyled__TabsWrapperStyled-kxWKwf kQdeyK"
>
  <div
    className="TabsContentWrapperStyled__TabsContentWrapperStyled-fmMkZg HDXly"
  >
    <div
      className="TabsListStyled__TabsListStyled-jltvsP iRzhNT"
    >
      <div
        className="TabStyled__TabStyled-kKgjJZ jFBgwR"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
        >
          <div
            className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
          >
            <div
              className="TabIconStyled__TabIconStyled-hvQHiJ kYPJWE"
              selected={false}
              type="svg"
            >
              <div
                className="IconSvgStyled__IconSvgStyled-iCtyLo hlSHcB"
                color=""
                rotate={0}
                size="normal"
                type="svg"
              >
                <svg
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="TabTextStyled__TabTextStyled-jbYAge hCrmvf"
            >
              Test 1
            </div>
          </div>
        </div>
      </div>
      <div
        className="TabStyled__TabStyled-kKgjJZ jfjewJ"
        onClick={[Function]}
        selected={true}
      >
        <div
          className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
        >
          <div
            className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
          >
            <div
              className="TabIconStyled__TabIconStyled-hvQHiJ fUMIYU"
              selected={true}
              type="svg"
            >
              <div
                className="IconSvgStyled__IconSvgStyled-iCtyLo hlSHcB"
                color=""
                rotate={0}
                size="normal"
                type="svg"
              >
                <svg
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="TabTextStyled__TabTextStyled-jbYAge hCrmvf"
            >
              Test 2
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly if prop  1`] = `
<div
  className="TabsWrapperStyled__TabsWrapperStyled-kxWKwf kQdeyK"
>
  <div
    className="TabsContentWrapperStyled__TabsContentWrapperStyled-fmMkZg HDXly"
  >
    <div
      className="TabsListStyled__TabsListStyled-jltvsP iRzhNT"
    >
      <a
        href="test1"
        onClick={[Function]}
      >
        <div
          className="TabStyled__TabStyled-kKgjJZ jfjewJ"
          onClick={[Function]}
          selected={true}
        >
          <div
            className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
          >
            <div
              className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
            >
              <div
                className="TabTextStyled__TabTextStyled-jbYAge cybLXN"
              >
                Test1
              </div>
            </div>
          </div>
        </div>
      </a>
      <a
        href="test2"
        onClick={[Function]}
      >
        <div
          className="TabStyled__TabStyled-kKgjJZ jFBgwR"
          onClick={[Function]}
          selected={false}
        >
          <div
            className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
          >
            <div
              className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
            >
              <div
                className="TabTextStyled__TabTextStyled-jbYAge cybLXN"
              >
                Test2
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly if prop icon 1`] = `
<div
  className="TabsWrapperStyled__TabsWrapperStyled-kxWKwf kQdeyK"
>
  <div
    className="TabsContentWrapperStyled__TabsContentWrapperStyled-fmMkZg HDXly"
  >
    <div
      className="TabsListStyled__TabsListStyled-jltvsP iRzhNT"
    >
      <div
        className="TabStyled__TabStyled-kKgjJZ jfjewJ"
        onClick={[Function]}
        selected={true}
      >
        <div
          className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
        >
          <div
            className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
          >
            <div
              className="TabIconStyled__TabIconStyled-hvQHiJ fUMIYU"
              selected={true}
              type="svg"
            >
              <div
                className="IconSvgStyled__IconSvgStyled-iCtyLo hlSHcB"
                color=""
                rotate={0}
                size="normal"
                type="svg"
              >
                <svg
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="TabTextStyled__TabTextStyled-jbYAge hCrmvf"
            >
              Test
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly if prop linkComponent passed 1`] = `
<div
  className="TabsWrapperStyled__TabsWrapperStyled-kxWKwf kQdeyK"
>
  <div
    className="TabsContentWrapperStyled__TabsContentWrapperStyled-fmMkZg HDXly"
  >
    <div
      className="TabsListStyled__TabsListStyled-jltvsP iRzhNT"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="TabStyled__TabStyled-kKgjJZ jfjewJ"
          onClick={[Function]}
          selected={true}
        >
          <div
            className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
          >
            <div
              className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
            >
              <div
                className="TabTextStyled__TabTextStyled-jbYAge cybLXN"
              >
                Test
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly if prop selected set 1 1`] = `
<div
  className="TabsWrapperStyled__TabsWrapperStyled-kxWKwf kQdeyK"
>
  <div
    className="TabsContentWrapperStyled__TabsContentWrapperStyled-fmMkZg HDXly"
  >
    <div
      className="TabsListStyled__TabsListStyled-jltvsP iRzhNT"
    >
      <div
        className="TabStyled__TabStyled-kKgjJZ jFBgwR"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
        >
          <div
            className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
          >
            <div
              className="TabTextStyled__TabTextStyled-jbYAge cybLXN"
            >
              Test
            </div>
          </div>
        </div>
      </div>
      <div
        className="TabStyled__TabStyled-kKgjJZ jfjewJ"
        onClick={[Function]}
        selected={true}
      >
        <div
          className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
        >
          <div
            className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
          >
            <div
              className="TabTextStyled__TabTextStyled-jbYAge cybLXN"
            >
              Test
            </div>
          </div>
        </div>
      </div>
      <div
        className="TabStyled__TabStyled-kKgjJZ jFBgwR"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
        >
          <div
            className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
          >
            <div
              className="TabTextStyled__TabTextStyled-jbYAge cybLXN"
            >
              Test
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly if prop selected set after render 1`] = `
<div
  className="TabsWrapperStyled__TabsWrapperStyled-kxWKwf kQdeyK"
>
  <div
    className="TabsContentWrapperStyled__TabsContentWrapperStyled-fmMkZg HDXly"
  >
    <div
      className="TabsListStyled__TabsListStyled-jltvsP iRzhNT"
    >
      <div
        className="TabStyled__TabStyled-kKgjJZ jFBgwR"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
        >
          <div
            className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
          >
            <div
              className="TabTextStyled__TabTextStyled-jbYAge cybLXN"
            >
              Test
            </div>
          </div>
        </div>
      </div>
      <div
        className="TabStyled__TabStyled-kKgjJZ jFBgwR"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
        >
          <div
            className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
          >
            <div
              className="TabTextStyled__TabTextStyled-jbYAge cybLXN"
            >
              Test
            </div>
          </div>
        </div>
      </div>
      <div
        className="TabStyled__TabStyled-kKgjJZ jfjewJ"
        onClick={[Function]}
        selected={true}
      >
        <div
          className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
        >
          <div
            className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
          >
            <div
              className="TabTextStyled__TabTextStyled-jbYAge cybLXN"
            >
              Test
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly if prop selected set after render 2`] = `
<div
  className="TabsWrapperStyled__TabsWrapperStyled-kxWKwf kQdeyK"
>
  <div
    className="TabsContentWrapperStyled__TabsContentWrapperStyled-fmMkZg HDXly"
  >
    <div
      className="TabsListStyled__TabsListStyled-jltvsP iRzhNT"
    >
      <div
        className="TabStyled__TabStyled-kKgjJZ jFBgwR"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
        >
          <div
            className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
          >
            <div
              className="TabTextStyled__TabTextStyled-jbYAge cybLXN"
            >
              Test
            </div>
          </div>
        </div>
      </div>
      <div
        className="TabStyled__TabStyled-kKgjJZ jfjewJ"
        onClick={[Function]}
        selected={true}
      >
        <div
          className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
        >
          <div
            className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
          >
            <div
              className="TabTextStyled__TabTextStyled-jbYAge cybLXN"
            >
              Test
            </div>
          </div>
        </div>
      </div>
      <div
        className="TabStyled__TabStyled-kKgjJZ jFBgwR"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
        >
          <div
            className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
          >
            <div
              className="TabTextStyled__TabTextStyled-jbYAge cybLXN"
            >
              Test
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly if prop text is empty 1`] = `
<div
  className="TabsWrapperStyled__TabsWrapperStyled-kxWKwf kQdeyK"
>
  <div
    className="TabsContentWrapperStyled__TabsContentWrapperStyled-fmMkZg HDXly"
  >
    <div
      className="TabsListStyled__TabsListStyled-jltvsP iRzhNT"
    >
      <div
        className="TabStyled__TabStyled-kKgjJZ jfjewJ"
        onClick={[Function]}
        selected={true}
      >
        <div
          className="TabContentWrapperStyled__TabContentWrapperStyled-drsnLo bNHOuk"
        >
          <div
            className="TabContentStyled__TabContentStyled-eLzkAQ bCjtic"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly with default props 1`] = `
<div
  className="TabsWrapperStyled__TabsWrapperStyled-kxWKwf kQdeyK"
>
  <div
    className="TabsContentWrapperStyled__TabsContentWrapperStyled-fmMkZg HDXly"
  >
    <div
      className="TabsListStyled__TabsListStyled-jltvsP iRzhNT"
    />
  </div>
</div>
`;
